### ISSUE TITLE
Docker version check is ignoring k8s version

### ISSUE THREAD

**Issue by maxkorp** [rancher/rke#931]
**RKE version: Latest (7b771410078bcb2f79f161fc2a1ab1d8294d9dba)**

```
nodes:
    - address: xxx.xxx.xxx.xxx
      internal_address: xxx.xxx.xxx.xxx
      user: ubuntu
      role:
        - controlplane
        - etcd
      ssh_key_path: /Users/maxkorp/.ssh/xxx.pem
    - address: xxx.xxx.xxx.xxx
      internal_address: xxx.xxx.xxx.xxx
      user: ubuntu
      role:
        - worker
      ssh_key_path: /Users/maxkorp/.ssh/xxx.pem
    - address: xxx.xxx.xxx.xxx
      internal_address: 
      user: ubuntu
      role:
        - worker
      ssh_key_path: /Users/maxkorp/.ssh/xxx.pem
kubernetes_version: "v1.12.0-rancher1-1"
```

**Steps to Reproduce:**
try to run `./rke up`

**Results:**
`Unsupported Docker version found [18.06.1-ce], supported versions are [1.11.x 1.12.x 1.13.x 17.03.x]`

Looking in `hosts/tunnel.go`, it appears `K8sVersion` is a const hardcoded to 1.8, rather than being pulled somewhere from the config.

**Comment by alena1108** [comment]
@moelsayed can there be an issue with the way we run k8s version - docker version comparison? As I see 1.18 being present in the list of supported versions for k8s 1.12: https://github.com/rancher/rke/blob/c4e77fbf3885d6f136584c205578cb6bf7cfc5a3/docker/docker.go#L41

**Comment by galal-hussein** [comment]
version: master

I was able to confirm that the issue was fixed with version `v1.12.0-rancher1` and other versions too.
